src/Nox.Lib/Factories/Types/NoxTypeDayOfWeekFactory.cs

```diff
@@ -0,0 +1,21 @@
+﻿using Nox.Solution;
+using Nox.Types;
+
+namespace Nox.Factories.Types
+{
+    public class NoxTypeDayOfWeekFactory : NoxTypeFactoryBase<Nox.Types.DayOfWeek>
+    {
+        public NoxTypeDayOfWeekFactory(NoxSolution solution) : base(solution)
+        {
+        }
+
+        public override Nox.Types.DayOfWeek? CreateNoxType(NoxSimpleTypeDefinition simpleTypeDefinition, dynamic? value)
+        {
+            if (value == null)
+            {
+                return null;
+            }
+            return Nox.Types.DayOfWeek.From(value);
```

Comment:
DayOfWeek currently is of type "int", can we maybe align the ValueObject primitive type to "ushort" and NoxType.cs to "ushort" 

![image](https://github.com/NoxOrg/Nox.Generator/assets/131167535/1c6ad08c-273c-4335-b39e-4ef1cadbf392)

![image](https://github.com/NoxOrg/Nox.Generator/assets/131167535/5a25f2bc-9124-481d-832d-6902a4a78d1d)


@rochar  can you please give your feedback here?

Reply:
agree, so we consolidate the Value object, the Converter and NoxType

Reply:
Base type updated.